src/rust/cryptography-x509-validation/src/policy/mod.rs

```diff
@@ -0,0 +1,130 @@
+        AlgorithmIdentifier {
+            oid: asn1::DefinedByMarker::marker(),
+            params: AlgorithmParameters::EcDsaWithSha256(None),
+        },
+        // For P-384: the signature MUST use ECDSA with SHA‐384
+        AlgorithmIdentifier {
+            oid: asn1::DefinedByMarker::marker(),
+            params: AlgorithmParameters::EcDsaWithSha384(None),
+        },
+        // For P-521: the signature MUST use ECDSA with SHA‐512
+        AlgorithmIdentifier {
+            oid: asn1::DefinedByMarker::marker(),
+            params: AlgorithmParameters::EcDsaWithSha512(None),
+        },
+    ])
+});
+
+#[cfg(test)]
+mod tests {
+    use super::WEBPKI_PERMITTED_ALGORITHMS;
+
+    fn decode_hex(s: &str) -> Vec<u8> {
+        (0..s.len())
+            .step_by(2)
+            .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
+            .collect::<Result<Vec<_>, _>>()
+            .unwrap()
+    }
+
+    #[test]
+    fn test_webpki_permitted_algorithms_canonical_encodings() {
+        // Expected encodings, from CA/B Forum's Baseline Requirements, section 7.1.3.2 (pages 96-98)
+        // https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-v2.0.0.pdf
+        let mut expected_encodings: Vec<_> = vec![
+            // RSASSA‐PKCS1‐v1_5 with SHA‐256
+            decode_hex("300d06092a864886f70d01010b0500"),
+            // RSASSA‐PKCS1‐v1_5 with SHA‐384
+            decode_hex("300d06092a864886f70d01010c0500"),
+            // RSASSA‐PKCS1‐v1_5 with SHA‐512
+            decode_hex("300d06092a864886f70d01010d0500"),
+            // RSASSA‐PSS with SHA‐256, MGF‐1 with SHA‐256, and a salt length of 32 bytes
+            decode_hex("304106092a864886f70d01010a3034a00f300d06096086480165030402010500a11c301a06092a864886f70d010108300d06096086480165030402010500a203020120"),
+            // RSASSA‐PSS with SHA‐384, MGF‐1 with SHA‐384, and a salt length of 48 bytes
+            decode_hex("304106092a864886f70d01010a3034a00f300d06096086480165030402020500a11c301a06092a864886f70d010108300d06096086480165030402020500a203020130"),
+            // RSASSA‐PSS with SHA‐512, MGF‐1 with SHA‐512, and a salt length of 64 bytes
+            decode_hex("304106092a864886f70d01010a3034a00f300d06096086480165030402030500a11c301a06092a864886f70d010108300d06096086480165030402030500a203020140"),
+            // ECDSA: P-256 with SHA256
+            decode_hex("300a06082a8648ce3d040302"),
+            // ECDSA: P-384 with SHA384
+            decode_hex("300a06082a8648ce3d040303"),
+            // ECDSA: P-521 with SHA512
+            decode_hex("300a06082a8648ce3d040304"),
+        ];
+        expected_encodings.sort();
+
+        let mut actual_encodings: Vec<_> = WEBPKI_PERMITTED_ALGORITHMS
+            .iter()
+            .map(|ai| asn1::write_single(ai).unwrap())
+            .collect();
+        actual_encodings.sort();
+
+        for (exp, act) in expected_encodings.iter().zip(actual_encodings.iter()) {
+            assert_eq!(act, exp);
+        }
```

Comment:
I think it'd make a lot more sense if each of the allowed algortihms was its own const, which we asserted had the correct encoding.